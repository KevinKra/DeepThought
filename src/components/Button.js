import React from "react";
import "../styles/_Button.scss";
class Button extends React.Component {
  constructor() {
    super();
    this.state = {};
  }
  buttonPath = () => {
    if (this.props.name === "Got it!") {
      this.props.handleResponse("GOT IT");
      console.log("button-detect got it");
    } else {
      this.props.handleResponse("MORE PRACTICE");
      console.log("button-detect more practice");
    }
  };
  render() {
    return (
      <button
        onClick={this.buttonPath}
        disabled={!this.props.submitStatus}
        className="btn-2"
      >
        {this.props.name}
      </button>
    );
  }
}

export default Button;
