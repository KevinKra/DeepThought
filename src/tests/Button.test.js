import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "../components/Button";

const defaultState = {};

describe("<Button />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Button />);
  });
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should match default state", () => {
    expect(wrapper.state()).toEqual(defaultState);
  });
  // it("should be disabled if submitStatus prop is false", () => {
  //test that submitStatus function is called
  // expect(wrapper.instance().props.submitStatus).toBe(true);

  //test that submitStatus results in button being disabled

  //onClick invoke buttonPath
  //   wrapper.find("button").simulate("click");
  // });
});
