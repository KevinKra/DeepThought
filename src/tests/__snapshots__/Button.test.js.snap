// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match the snapshot 1`] = `
<button
  className="btn-2"
  disabled={true}
  onClick={[Function]}
/>
`;
